docs/decisions/0003-2FA-for-course-staff-accounts.rst

```diff
@@ -0,0 +1,155 @@
+
+This change can be accommodated within our current login flow. A high level flow of this process is as follow:
+
+- A user belonging to a course team will authenticate themselves like they currently do, either by providing login credentials or by SSO.
+- Once they have authenticated successfully they will be authenticated by a 2FA method
+- After successful verification of 2FA code, assign a session to user.
+
+Why choose 2FA?
+***************
+
+- It is a conventional way to secure user accounts.
+- We can protect staff users at the first step they take on our site i.e authentication.
+- Development wise it will not require extensive effort as compared to the second option.
+
+
+Two Factor Authentication approaches
+************************************
+
+- Email Token
+- Sms Token
+- Push Based 2FA
+- TOTP
+- U2F Security Keys
+
+
+Comparison
+**********
+
+.. list-table:: **Comparison of 2FA methods**
+   :widths: 10 15 15 15
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Types
+     - Intro
+     - Pros
+     - Cons
+   * - Email Token
+     - Send a token to the registered email address and verify the token, during login.
+     - - Works fine if the email is secured
+       - Extra device not needed
+       - We already have user email so we will not be requiring any personal information from the user.
+     - - Vulnerable if password is reused. (most users keep same password)
+       - Vulnerable if device is compromised (malware attack)
+       - Not secured as other 2FA methods.
+       - Email delivery concern.
+   * - SMS Token
+     - Send a token to the registered mobile number and verify the token, during login.
+     - - No need to install extra app
+       - Works fine if mobile network is not ported
+     - - Vulnerable to targeted attacks like social engineering and number porting etc.
+       - Requires a cell phone.
+       - Privacy concern (people not comfortable giving phone number)
+       - Cannot use if phone is dead or unable to reach signals
+       - We will be getting a mobile phone number ( a new info) so we will be needing compliance approval.
+   * - Push Based
+     - Send a push notification on your trusted device, the prompt will ask for accept and denial.
+     - - Secured
+       - Convenient
+       - Time saving
+     - - Cannot use single app for multiple sites
+       - Requires signal and connection
+       - Cannot use if phone is dead
+       - Requires custom integrations from providers.
+   * - U2F
+     - Using small USB, NFC or Bluetooth Low Energy (BTLE) devices often called “security keys.” To set it up on a site, you register your U2F device. And then tap on the device when logging in.
+     - - Secured
+       - Can use same U2F device for multiple sites
+     - - Needs a device
+       - Cost
+       - Support (only chrome supports U2F currently)
```

Comment:
Support is broader than that -- Firefox supports U2F as well. (But agreed that it can't be the only supported method, so probably not something we want to start with.)